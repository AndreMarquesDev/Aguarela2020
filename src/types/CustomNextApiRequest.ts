import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
