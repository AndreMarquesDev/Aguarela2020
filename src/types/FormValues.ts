export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
