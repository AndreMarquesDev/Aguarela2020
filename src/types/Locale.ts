export enum Locale {
    Pt = 'pt',
    En = 'en',
}
