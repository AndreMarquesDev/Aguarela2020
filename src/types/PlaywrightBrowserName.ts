export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
