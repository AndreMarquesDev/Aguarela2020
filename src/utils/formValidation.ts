export enum FieldTypes {
    Name = 'name',
    Email = 'email',
    Textarea = 'message',
    Brand = 'brand',
    Subject = 'subject',
}
export type FormState = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};

export const validateName = (value: string): boolean => {
    const nameRegex =
        // eslint-disable-next-line regexp/no-unused-capturing-group, regexp/prefer-plus-quantifier, regexp/prefer-character-class, regexp/no-super-linear-backtracking, regexp/no-useless-escape
        /s*([a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]{1,}([\.,] |[-']| ))+[a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]+\.?\s*$/;

    return nameRegex.test(value);
};

export const validateEmail = (value: string): boolean => {
    const emailRegex =
        // eslint-disable-next-line regexp/no-useless-escape, regexp/no-unused-capturing-group, regexp/prefer-d, regexp/use-ignore-case
        /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    return emailRegex.test(value);
};
