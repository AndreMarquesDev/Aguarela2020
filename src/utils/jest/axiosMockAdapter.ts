import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { contactFormBackendUrl } from '../urls';

export const initializeAxiosMockAdapter = (delay = 0, isSuccess = true): void => {
    const mock = new MockAdapter(axios, {
        delayResponse: delay,
    });

    const httpStatusCode = isSuccess ? 200 : 500;

    mock.onPost(contactFormBackendUrl).reply(httpStatusCode, {});
};
