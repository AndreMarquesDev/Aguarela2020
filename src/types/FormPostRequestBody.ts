import { FormValues } from './FormValues';

export interface FormPostRequestBody extends FormValues {
    isPlaywright: boolean;
}
