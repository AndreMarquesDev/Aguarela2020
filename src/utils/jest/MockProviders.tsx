/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
