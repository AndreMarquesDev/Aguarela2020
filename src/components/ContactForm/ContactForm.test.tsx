import '@testing-library/jest-dom';
import React from 'react';
import {
    render,
    RenderResult,
    screen,
    waitFor,
    waitForElementToBeRemoved,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ContactForm, FORM_RESET_TIMEOUT } from './ContactForm';
import { textsEn, textsPt } from '../../utils/texts';
import { FieldTypes } from '../../utils/formValidation';
import { initializeAxiosMockAdapter } from '../../utils/jest/axiosMockAdapter';
import {
    contactFormContainerDataTestId,
    contactFormErrorMessageHiddenDataTestId,
    contactFormErrorMessageVisibleDataTestId,
} from '../../utils/dataTestIds';
import { MockProviders } from '../../utils/jest/MockProviders';
import { Locale } from '../../types/Locale';

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <ContactForm />
        </MockProviders>
    );
};

describe('<ContactForm />', () => {
    test('renders properly', () => {
        renderComponent();

        const hiddenErrorMessages = screen.getAllByTestId(contactFormErrorMessageHiddenDataTestId);

        expect(hiddenErrorMessages.length).toBe(5);
        expect(screen.getByText(textsPt.send)).toBeInTheDocument();

        expect(screen.getByTestId(contactFormContainerDataTestId)).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        renderComponent(Locale.En);

        const hiddenErrorMessages = screen.getAllByTestId(contactFormErrorMessageHiddenDataTestId);

        expect(hiddenErrorMessages.length).toBe(5);
        expect(screen.getByText(textsEn.send)).toBeInTheDocument();

        expect(screen.getByTestId(contactFormContainerDataTestId)).toMatchSnapshot();
    });

    test('displays error messages in required fields on submit with empty fields', async () => {
        renderComponent();

        const submitButton = screen.getByText(textsPt.send);

        userEvent.click(submitButton);

        const errorMessages = await screen.findAllByTestId(
            contactFormErrorMessageVisibleDataTestId
        );

        expect(errorMessages.length).toBe(3);

        expect(screen.getByTestId(contactFormContainerDataTestId)).toMatchSnapshot();
    });

    test('lets fields be properly filled', async () => {
        renderComponent();

        const nameInput = screen.getByTestId(FieldTypes.Name);
        const brandInput = screen.getByTestId(FieldTypes.Brand);
        const emailInput = screen.getByTestId(FieldTypes.Email);
        const subjectInput = screen.getByTestId(FieldTypes.Subject);
        const textareaField = screen.getByTestId(FieldTypes.Textarea);

        userEvent.type(nameInput, 'André{space}Marques');
        userEvent.type(brandInput, 'Brand');
        userEvent.type(emailInput, 'email@email.com');
        userEvent.type(subjectInput, 'Subject');
        userEvent.type(textareaField, 'Textarea{space}field');

        await waitFor(() => {
            // eslint-disable-next-line testing-library/no-wait-for-snapshot
            expect(screen.getByTestId(contactFormContainerDataTestId)).toMatchSnapshot();
        });
    });

    test('submits the form successfully', async () => {
        initializeAxiosMockAdapter();

        renderComponent();

        const nameInput = screen.getByTestId(FieldTypes.Name);
        const brandInput = screen.getByTestId(FieldTypes.Brand);
        const emailInput = screen.getByTestId(FieldTypes.Email);
        const subjectInput = screen.getByTestId(FieldTypes.Subject);
        const textareaField = screen.getByTestId(FieldTypes.Textarea);

        userEvent.type(nameInput, 'André{space}Marques');
        userEvent.type(brandInput, 'Brand');
        userEvent.type(emailInput, 'email@email.com');
        userEvent.type(subjectInput, 'Subject');
        userEvent.type(textareaField, 'Textarea{space}field');

        const submitButton = screen.getByText(textsPt.send);

        userEvent.click(submitButton);

        const container = await screen.findByTestId('contactForm_container');

        expect(container).toMatchSnapshot();
    });

    test(
        'resets the form after it was successfully submitted',
        async () => {
            initializeAxiosMockAdapter();

            renderComponent();

            const nameInput = screen.getByTestId(FieldTypes.Name);
            const brandInput = screen.getByTestId(FieldTypes.Brand);
            const emailInput = screen.getByTestId(FieldTypes.Email);
            const subjectInput = screen.getByTestId(FieldTypes.Subject);
            const textareaField = screen.getByTestId(FieldTypes.Textarea);

            userEvent.type(nameInput, 'André{space}Marques');
            userEvent.type(brandInput, 'Brand');
            userEvent.type(emailInput, 'email@email.com');
            userEvent.type(subjectInput, 'Subject');
            userEvent.type(textareaField, 'Textarea{space}field');

            const submitButton = screen.getByText(textsPt.send);

            userEvent.click(submitButton);

            const container = await screen.findByTestId('contactForm_container');

            await waitForElementToBeRemoved(screen.queryByText('Mensagem enviada com sucesso.'), {
                timeout: FORM_RESET_TIMEOUT,
            });

            expect(container).toMatchSnapshot();
        },
        FORM_RESET_TIMEOUT + 1000
    );

    test('shows an error message in case the POST request fails', async () => {
        initializeAxiosMockAdapter(0, false);

        renderComponent();

        const nameInput = screen.getByTestId(FieldTypes.Name);
        const brandInput = screen.getByTestId(FieldTypes.Brand);
        const emailInput = screen.getByTestId(FieldTypes.Email);
        const subjectInput = screen.getByTestId(FieldTypes.Subject);
        const textareaField = screen.getByTestId(FieldTypes.Textarea);

        userEvent.type(nameInput, 'André{space}Marques');
        userEvent.type(brandInput, 'Brand');
        userEvent.type(emailInput, 'email@email.com');
        userEvent.type(subjectInput, 'Subject');
        userEvent.type(textareaField, 'Textarea{space}field');

        const submitButton = screen.getByText(textsPt.send);

        userEvent.click(submitButton);

        const container = await screen.findByTestId('contactForm_container');

        expect(container).toMatchSnapshot();
    });
});
