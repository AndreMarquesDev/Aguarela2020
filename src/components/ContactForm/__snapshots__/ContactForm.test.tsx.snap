// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> displays error messages in required fields on submit with empty fields 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Nome
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="true"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage visible"
          data-testid="errorMessage_visible"
          id="nameErrorMessage"
        >
          Este campo é obrigatório. Por favor insira o seu primeiro e último nome
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Marca/Negócio
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="true"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage visible"
          data-testid="errorMessage_visible"
          id="emailErrorMessage"
        >
          Este campo é obrigatório. Por favor insira um endereço de email válido
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Assunto
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Mensagem
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="true"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        />
        <span
          class="jsx-1980812632 errorMessage visible"
          data-testid="errorMessage_visible"
          id="messageErrorMessage"
        >
          Este campo é obrigatório. Por favor insira uma mensagem
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`<ContactForm /> lets fields be properly filled 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Nome
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value="André Marques"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="nameErrorMessage"
        >
          Este campo é obrigatório. Por favor insira o seu primeiro e último nome
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Marca/Negócio
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value="Brand"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value="email@email.com"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="emailErrorMessage"
        >
          Este campo é obrigatório. Por favor insira um endereço de email válido
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Assunto
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value="Subject"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Mensagem
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        >
          Textarea field
        </textarea>
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="messageErrorMessage"
        >
          Este campo é obrigatório. Por favor insira uma mensagem
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`<ContactForm /> renders properly 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Nome
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="nameErrorMessage"
        >
          Este campo é obrigatório. Por favor insira o seu primeiro e último nome
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Marca/Negócio
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="emailErrorMessage"
        >
          Este campo é obrigatório. Por favor insira um endereço de email válido
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Assunto
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Mensagem
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="messageErrorMessage"
        >
          Este campo é obrigatório. Por favor insira uma mensagem
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`<ContactForm /> renders properly in English 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Name
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="nameErrorMessage"
        >
          This field is required. Please enter your first and last name
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Brand/Business
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="emailErrorMessage"
        >
          This field is required. Please enter a valid email address
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Subject
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Message
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="messageErrorMessage"
        >
          This field is required. Please enter a message
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Send
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`<ContactForm /> resets the form after it was successfully submitted 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Nome
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="nameErrorMessage"
        >
          Este campo é obrigatório. Por favor insira o seu primeiro e último nome
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Marca/Negócio
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="emailErrorMessage"
        >
          Este campo é obrigatório. Por favor insira um endereço de email válido
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Assunto
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value=""
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Mensagem
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="messageErrorMessage"
        >
          Este campo é obrigatório. Por favor insira uma mensagem
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`<ContactForm /> shows an error message in case the POST request fails 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Nome
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value="André Marques"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="nameErrorMessage"
        >
          Este campo é obrigatório. Por favor insira o seu primeiro e último nome
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Marca/Negócio
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value="Brand"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value="email@email.com"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="emailErrorMessage"
        >
          Este campo é obrigatório. Por favor insira um endereço de email válido
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Assunto
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value="Subject"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Mensagem
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="true"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        >
          Textarea field
        </textarea>
        <span
          class="jsx-1980812632 errorMessage visible"
          data-testid="errorMessage_visible"
          id="messageErrorMessage"
        >
          Este campo é obrigatório. Por favor insira uma mensagem
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter disabled"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </div>
      <span
        class="jsx-3940016528 formNotification error"
        id="contactForm"
      >
        Ocorreu um erro no envio da mensagem. Tente outra vez mais tarde ou entre em contacto através do meu email 
        <a
          class="link"
          href="mailto:catarina@aguareladigital.com"
        >
          catarina@aguareladigital.com
        </a>
      </span>
    </form>
  </div>
</section>
`;

exports[`<ContactForm /> submits the form successfully 1`] = `
<section
  class="jsx-3940016528 container"
  data-testid="contactForm_container"
>
  <div
    class="jsx-3940016528 wrapper genericMargins"
  >
    <form
      aria-describedby="contactForm"
      class="jsx-3940016528"
      novalidate=""
    >
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="name"
        >
          Nome
           *
        </label>
        <input
          aria-describedby="nameErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="name"
          id="name"
          name="name"
          required=""
          title="name"
          type="text"
          value="André Marques"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="nameErrorMessage"
        >
          Este campo é obrigatório. Por favor insira o seu primeiro e último nome
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="brand"
        >
          Marca/Negócio
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="brand"
          id="brand"
          name="brand"
          title="brand"
          type="text"
          value="Brand"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="email"
        >
          Email
           *
        </label>
        <input
          aria-describedby="emailErrorMessage"
          class="jsx-1980812632"
          data-error="false"
          data-testid="email"
          id="email"
          name="email"
          required=""
          title="email"
          type="email"
          value="email@email.com"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
          id="emailErrorMessage"
        >
          Este campo é obrigatório. Por favor insira um endereço de email válido
        </span>
      </div>
      <div
        class="jsx-1980812632 fieldWrapper"
      >
        <label
          class="jsx-1980812632"
          for="subject"
        >
          Assunto
        </label>
        <input
          class="jsx-1980812632"
          data-error="false"
          data-testid="subject"
          id="subject"
          name="subject"
          title="subject"
          type="text"
          value="Subject"
        />
        <span
          class="jsx-1980812632 errorMessage"
          data-testid="errorMessage_hidden"
        />
      </div>
      <div
        class="jsx-1980812632 fieldWrapper textarea"
      >
        <label
          class="jsx-1980812632"
          for="message"
        >
          Mensagem
           *
        </label>
        <textarea
          aria-describedby="messageErrorMessage"
          class="jsx-1980812632"
          data-error="true"
          data-testid="message"
          id="message"
          name="message"
          required=""
          title="message"
        >
          Textarea field
        </textarea>
        <span
          class="jsx-1980812632 errorMessage visible"
          data-testid="errorMessage_visible"
          id="messageErrorMessage"
        >
          Este campo é obrigatório. Por favor insira uma mensagem
        </span>
      </div>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter disabled"
        >
          <button
            class="jsx-2566809609 button"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </div>
      <span
        class="jsx-3940016528 formNotification"
        id="contactForm"
      >
        Mensagem enviada com sucesso.
      </span>
    </form>
  </div>
</section>
`;
